### PR TITLE
Veteran#find_or_create_poa is now a thing

### PR DESCRIPTION
**Pull request by n1zyy**
### Description
On Bat Team, we very, very often run:

```ruby
v = appeal.veteran
poa = BgsPowerOfAttorney.find_or_create_by_file_number(v.file_number)
```

This does not lend itself to muscle memory. It also feels like a Veteran object should know how to retrieve its POA.

This is mostly meant as a convenience for people in the console, not the "official" way to find a POA for a Veteran.  I didn't add any tests to avoid promoting it as the canonical way to find a POA.

Now you can run:
```ruby
v = appeal.veteran
poa = v.find_or_create_poa
```

The implement is literally just the `BgsPowerOfAttorney.find_or_create_by_file_number` put in the model.

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] People agree this would be convenient, rather than a weird hack only I would ever use

### Testing Plan
This can be tested pretty simply in the console:

```
v = Veteran.last
poa = v.find_or_create_poa
```

### CODE REVIEW

**Review by jcq — APPROVED** [review]
Seems good to me, but I'd def appreciate others more opinionated about Ruby/Rails conventions to weigh in on convenience methods like this. 

You know, like the backend functional lead or something 😉

**Review by eileen-nava — APPROVED** [review]
This looks good to me. 👍

**Review by yoomlam — CHANGES_REQUESTED** [review]
Please add a corresponding test to maintain high test coverage

File: spec/models/veteran_spec.rb [hunk 1]
```diff
@@ -929,4 +929,13 @@
...
   end
+
+  describe "#find_or_create_poa" do
+    subject { veteran.find_or_create_poa }
+
+    it "calls BgsPowerOfAttorney" do
+      expect(BgsPowerOfAttorney).to receive(:find_or_create_by_file_number).with(veteran.file_number)
+      subject
+    end
+  end
```
Comment (n1zyy) [inline spec/models/veteran_spec.rb:940]: This test is intentionally a little bit sparse: the method only exists to call `BgsPowerOfAttorney.find_or_create_by_file_number(file_number)` in a more convenient way, so we simply test that that's what happens.

`BgsPowerOfAttorney.find_or_create_by_file_number` is already tested separately: https://github.com/department-of-veterans-affairs/caseflow/blob/5d0bf14c9747fc9440f70afcdcebe8a38c3f9a8e/spec/models/bgs_power_of_attorney_spec.rb#L69-L129

**Comment by n1zyy** [comment]
@yoomlam - Good call. Added! 👍

**Comment by n1zyy** [comment]
@jcq 

> Seems good to me, but I'd def appreciate others more opinionated about Ruby/Rails conventions to weigh in on convenience methods like this.

Yang had some good input here, on how there's a [console_methods.rb helper](https://github.com/department-of-veterans-affairs/caseflow/blob/5d0bf14c9747fc9440f70afcdcebe8a38c3f9a8e/lib/helpers/console_methods.rb). But, talking it over with Team Victor which is working with this code the most, there was some discussion that perhaps this _should_ be the way to fetch a POA.

**Comment by yoomlam** [comment]
@n1zyy What do you want to do with this PR?

**Pull request closed without merging**